src/Func.cc

```diff
@@ -795,6 +790,9 @@ ValPtr BuiltinFunc::Invoke(Args* args, Frame* parent) const
 		g_trace_state.LogTrace("\tFunction return: %s\n", d.Description());
 		}
 
+	if ( spm )
```

Comment:
Just putting this out as food-for-thought if there's an opportunity to expose StartInvocation/EndInvocation to plugins.

There is at least one plugin that tracks function runtime via HookCallFunction and exposes it through a prometheus interface:
https://github.com/esnet/zeek-exporter/blob/master/src/Plugin.cc#L106

A recent MR around OpenTelemetry also hooks into Invoke():
https://github.com/zeek/zeek/pull/1998/files#diff-bc6dffe506a4add540a9b1df23ca34794aed8eec4ace91eeae9a9c7c12193e6aR396

Generally, seems if external plugins could hook into start and end invocation of functions, it could foster some interesting experiments around performance collection outside of the code base.

I was initially wondering if two new Hooks would be reasonable, but maybe the `spm` could represent a collection of `ScriptProfileMgr` instances (interface being just the `StartInvocation` `EndInvocation` methods) that Plugins can implement and register/deregister at will (possibly even during runtime).

Just thinking out loud. Maybe better suited for the discussions section.

Reply:
Yeah I can see that being handy ... would prefer though to get this simpler approach integrated for now since I'd like to have the functionality in 5.0.